ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -4329,4 +4367,28 @@ mod tests {
             ",
         );
     }
+
+    #[test]
+    fn cg_fneg() {
+        codegen_and_test(
+            "
+              entry:
+                %0: float = load_ti 0
+                %1: double = load_ti 1
+                %2: float = fneg %0
+                %3: double = fneg %1
+            ",
+            "
+                ...
+                ; %2: float = fneg %0
+                {{_}} {{_}}: mov r.32.x, 0x80000000
+                {{_}} {{_}}: movd fp.128.y, r.32.x
+                {{_}} {{_}}: xorps fp.128.z, fp.128.y
+                ; %3: double = fneg %1
+                {{_}} {{_}}: mov r.64.x, 0x8000000000000000
+                {{_}} {{_}}: movq fp.128.y, r.64.x
+                {{_}} {{_}}: xorpd fp.128.a, fp.128.y
```

Comment:
I was curious that `%3` was allocated a fresh register (`fp.128.a`), instead of reusing the register previously assigned to the now-dead `%2` (`fp.128.z`). Could indicate an error in my register allocator changes?

Reply:
We currently prefer to allocate "higher" registers, which is pointless in FP-land, but in GP-land means we're less likely to get twonked by x64's fondness for RAX and the sysv abi's preference for "lower" registers in CALLs. Does that *guarantee* what you're seeing is intended? Nope. But it might give you a clue to what look for.